std/typecons.d

```diff
@@ -80,40 +79,69 @@ public:
     ---
     */
     static Unique!T create(A...)(auto ref A args)
-    if (__traits(compiles, new T(args)))
     {
+        import core.memory : GC;
+        import core.stdc.stdlib : malloc;
+        import std.conv : emplace;
+        import std.exception : enforce;
+
         debug(Unique) writeln("Unique.create for ", T.stringof);
         Unique!T u;
-        u._p = new T(args);
+
+        // TODO: May need to fix alignment?
+        // Does emplace still need to mess with alignment if
+        // the memory is coming from malloc, or does malloc handle that?
+
+        static if (is(T == class))
+            immutable size_t allocSize = __traits(classInstanceSize, T);
+        else
+            immutable size_t allocSize = T.sizeof;
+
+        void* rawMemory = enforce(malloc(allocSize), "malloc returned null");
+        u._p = cast(RefT)rawMemory;
```

Comment:
I appear to be having a segfault on classes that have a reference to their context, such as the one on the third unit test, on line 281. When I step through the constructor in GDB, `rawMemory` is correctly set to our newly allocated memory, but after stepping past this line, `u._p` is some impossibly small address, such as 0x14. It's like the assignment is ignored or the cast messes with the address somehow. Is this some failing of my understanding? The D site says that "Casting a pointer type to and from a class type is done as a type paint (i.e. a reinterpret cast)." so I'm not sure how this is happening.